nifi-ngsild-ckan-processors/src/main/java/egm/io/nifi/processors/ckan/utils/CKANAggregator.java

```diff
@@ -290,19 +291,31 @@ public void aggregateInitialAttributes(String attributePrefix, ArrayList<Attribu
         }
     }
 
-    public void aggregateAttributeValues(String attributePrefix, ArrayList<AttributesLD> attributes, LinkedHashMap<String, ArrayList<JsonElement>> aggregation) {
+    public void aggregateAttributeValues(String attributePrefix, ArrayList<AttributesLD> attributes, LinkedHashMap<String, ArrayList<JsonElement>> aggregation, String datasetIdPrefixToTruncate ) {
         for (AttributesLD attribute : attributes) {
-            String attrName = (attributePrefix.isEmpty() ? attributePrefix : attributePrefix + "_") + attribute.getAttrName();
+            String encodedAttrName = encodeAttributeName(attribute.getAttrName(), attribute.getDatasetId(), datasetIdPrefixToTruncate);
+            String attrName = (attributePrefix.isEmpty() ? attributePrefix : attributePrefix + "_") + encodedAttrName;
```

Comment:
if `attributePrefix` and `datasetId` are not empty, we will end up with a double `_` in the encoded attribute name